internal/oobmigration/version.go

```diff
@@ -14,6 +19,19 @@ func NewVersion(major, minor int) Version {
 	}
 }
 
+var versionPattern = lazyregexp.New(`^v?(\d+)\.(\d+)(?:\.\d+)?$`)
+
+func NewVersionFromString(v string) (Version, bool) {
+	if matches := versionPattern.FindStringSubmatch(v); len(matches) >= 3 {
+		major, _ := strconv.Atoi(matches[1])
+		minor, _ := strconv.Atoi(matches[2])
```

Comment:
```suggestion
		minor, err := strconv.Atoi(matches[2])
```

Not sure if you want to handle the errors by returning `false` here.

Reply:
It can't be non-nil for any reasonable versions if it passes the regex.